_posts/2021-02-03-#20721.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: pr
+date: 2021-02-03
+title: "Move ping data to net_processing"
+pr: 20721
+authors: [jnewbery]
+components: ["p2p"]
+host: jnewbery
+status: upcoming
+commit:
+---
+
+_Notes and questions to follow soon!_
+
+## Background
+
+- We can think of the Bitcoin Core node as a stack with three levels:
+
+  - **Net** - keeps track of potential nodes to peer with, handles active
+    connections to nodes, and reads and writes messages to the wire. The main
+    class in the net layer is `CConnman`, which is defined in
+    [net.cpp](https://github.com/bitcoin/bitcoin/blob/16b784d9/src/net.h). Net
+    hands messages up the stack to ...
+
+  - **Net Processing** - reads and processes application-level messages, and
+    handles application-level view of peers. The main class in net
+    processing is `PeerManager`, which is declared in
+    [net_processing.h](https://github.com/bitcoin/bitcoin/blob/16b784d9/src/net_processing.h).
+    Net processing uses function calls like `ProcessNewBlock()` and
+    `AcceptToMemoryPool()` to call into ...
+
+  - **Validation** - is responsible for our view of the block chain and the UTXO
+    set. It may (or may not) include a mempool for unconfirmed transactions. There
+    are various classes and functions exposed to the rest of the node. [PR
+    20158](https://github.com/bitcoin/bitcoin/pull/20158) and linked PRs are an
+    attempt to clarify and rationlize the interface to validation.
+
+- In early versions of Bitcoin, this distinction wasn't as clear. The [first
```

Comment:
maybe s/Bitcoin/Bitcoin Core/

Reply:
The software wasn't renamed "Bitcoin Core" until around 2013 if I recall correctly.